kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/system/SystemServiceImpl.java

```diff
@@ -573,8 +573,9 @@ public String getPrimaryMacAddress() {
 				Enumeration<NetworkInterface> networks = NetworkInterface.getNetworkInterfaces();
 				while (networks.hasMoreElements()) {
 					NetworkInterface network = networks.nextElement();
-					if(network.getIndex() == 0) {
+					if(network.getName().equals("eth0")) {
 						ip = network.getInetAddresses().nextElement();
```

Comment:
@malikm As a matter of style we encourage using:
``"eth0".equals()`` which will not throw NPE if ``network.getName()`` happens to return ``null``. Can you please rework it?